tests/utils.go

```diff
@@ -423,6 +440,183 @@ func CreateSecret(name string, data map[string]string) {
 	}
 }
 
+func GetListOfManifests(pathToManifestsDir string) []string {
+	var manifests []string
+	isOpenshift := IsOpenShift()
+	matchFileName := func(pattern, filename string) bool {
+		match, err := filepath.Match(pattern, filename)
+		if err != nil {
+			panic(err)
+		}
+		return match
+	}
+	err := filepath.Walk(pathToManifestsDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			fmt.Printf("ERROR: Can not access a path %q: %v\n", path, err)
+			return err
+		}
+		if !info.IsDir() {
+			if matchFileName("*-for-ocp.yaml", info.Name()) {
+				if isOpenshift {
+					manifests = append(manifests, path)
+				}
+			} else if matchFileName("*-for-k8s.yaml", info.Name()) {
+				if !isOpenshift {
+					manifests = append(manifests, path)
+				}
+			} else if matchFileName("*.yaml", info.Name()) {
+				manifests = append(manifests, path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Printf("ERROR: Walking the path %q: %v\n", pathToManifestsDir, err)
+		panic(err)
+	}
+	return manifests
+}
+
+func ReadManifestYamlFile(pathToManifest string) []unstructured.Unstructured {
+	var objects []unstructured.Unstructured
+	stream, err := os.Open(pathToManifest)
+	PanicOnError(err)
+
+	buff := make([]byte, 1024)
+	yamlDocDecoder := yaml.NewDocumentDecoder(stream)
```

Comment:
What you get back here is a "stream", you can do something like this:

```golang
reader :=  yaml.NewDocumentDecoder(stream)
decoder := NewYAMLOrJSONDecoder(reader, buff)
for {
   obj := map[string]interface{}{}
   err := decoder.Decode(obj)
   if err == io.EOF {
     break
  } else if err != nil {
    panic(err)
  }
  objects = append(objects, unstructured.Unstructured{Object: obj})
}
return objects
```

Reply:
It was about missing new line between documents in yaml files :-/